enhancements/console/user-settings.md

```diff
@@ -0,0 +1,134 @@
+
+Each user settings `ConfigMap` will have a generated name using `metadata.generateName: user-settings-`.
+
+In order to locate the `ConfigMap` associated to the user, labels will be used:
+
+- `console.openshift.io/user-settings: true`: identifies the `ConfigMap` for the purpose of user settings
+- `console.openshift.io/user: <user UID>`: associates the `ConfigMap` with its user by their unique identifier
+
+Requires a nhe new role binding for the console service account to grants access to `create`, `update`, and `patch` for `ConfigMap` resources in the chosen namespace.
+
+### Backend API
+
+The Console backend will handle all user settings requests through `/api/console/user-settings`. Since the user cannot directly create or update the `ConfigMap` resource, the backend will need to first fetch or create the users' `ConfigMap` in order to identify and update the correct resource.
+
+Initially support methods: `GET`, `PUT`, `PATCH`
+Depending on how settings are used by some features, we may also require support for watching the resource for changes.
+
+The backend service will lazily create exactly one `ConfigMap` per user. The Console frontend will query the backend service to fetch the contents of the `ConfigMap` when needed.
+
+### Settings Keys
+
+While the Console frontend will be the only consumer and producer of user settings, many features of the Console will contribute to user settings. This includes future dynamic contributions through extensibility. Therefore it is important to follow conventions for key creation in order to avoid collisions.
+
+Settings keys should be uniquely qualified using reverse domain notation:
+
+- `console.homepage`
+- `dev-console.topology.filters`
+
+### User Stories
+
+#### Story 1
+
+As a user, I want my guided tours progress to be saved with my account, such that returning to the Console on another machine will show my progress through each tour.
+
+### Risks and Mitigations
+
+**Risk**: Constraining user settings to only be accessible by their intended user.
+
+**Mitigation**: RBAC rule to only allow the console service account access to read and write the user settings and all user requests proxied through the Console backend.
+
+**Risk**: User settings keys between features may collide.
+
+**Mitigation**: Enforce qualified keys where possible and document best practice for creating unique keys.
+
+## Design Details
+
+## Open Questions
+
+- [ ] Is `ConfigMap` the best option, or a new `ConsoleUserSettings` CRD?
+- [ ] Which namespace to store user settings `ConfigMap` resources?
+- [ ] How to handle cleaning up stale user preferences?
+
+### Test Plan
+
+Testing will be carried out with the usual Console e2e and unit test suites.
+
+### Graduation Criteria
+
+None
+
+### Upgrade / Downgrade Strategy
+
+Support for this feature requires no special handling during upgrade. Default user settings will be provided.
```

Comment:
Do we need to migrate the existing settings from `localStorage`?

Reply:
We can merge the settings we store today in local storage.